### PR TITLE
Added a function to filter the installed packs in nightly and in a build

### PR DESCRIPTION
**Pull request by abaumgarten**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/33223

## Description
Added a function that will be responsible for filtering the installed packs by the same logic as happens now, but with a change - this function will not ignore contributed packs. This new function will be called in the nightly flow and in the build flow as well.
note: we keep the logic of filtering tests the same, i.e we will continue ignoring contributed packs.

## Screenshots
A successful build after changes in 3 packs from different support levels: 
1. xsoar - `AWS - GuardDuty`
2. partner - `Accessdata`
3. community - `Aella Star Light`

Screen Shot 2021-02-07 at 18 00 04
https://app.circleci.com/pipelines/github/demisto/content/66388/workflows/83577cf2-f036-4b87-8e3b-6569eb0cfeb5/jobs/274055

trigger nightly build on my branch:
https://app.circleci.com/pipelines/github/demisto/content/66520/workflows/5780e1a0-feb7-40ca-b2b3-796a64d69dcc

### CODE REVIEW

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 1]
```diff
@@ -1030,6 +1029,21 @@ def filter_tests(tests: set, id_set: json) -> set:
     return tests_without_non_supported
 
 
+def filter_installed_packs(packs_to_install: set) -> set:
+    """
+    Filter packs that should be installed out from the installed packs set if they are:
```
Comment (ShahafBenYakir) [inline Tests/scripts/collect_tests_and_content_packs.py:1034]:
```suggestion
    Filter only the packs that should get installed by the following conditions:
```

File: Tests/scripts/utils/tests/content_packs_util_test.py [hunk 2]
```diff
@@ -104,3 +105,44 @@ def test_should_test_content_pack(mocker, tmp_path, pack_metadata_content, pack_
...
+    """
+    Given:
+        - Case A: Partner content pack
+        - Case B: Community content pack
+        - Case C: NonSupported content pack
+        - Case D: Deprecated CortexXDR content pack
+        - Case E: ApiModules content pack
+
+    When:
+        - Checking if pack should be tested
```
Comment (ShahafBenYakir) [inline Tests/scripts/utils/tests/content_packs_util_test.py:131]:
```suggestion
        - Checking if pack should be installed
```

File: Tests/scripts/utils/tests/content_packs_util_test.py [hunk 3]
```diff
@@ -104,3 +105,44 @@ def test_should_test_content_pack(mocker, tmp_path, pack_metadata_content, pack_
...
+
+    Then:
+        - Case A: Verify pack should be installed
+        - Case B: Verify pack should not be installed
```
Comment (ShahafBenYakir) [inline Tests/scripts/utils/tests/content_packs_util_test.py:135]:
```suggestion
        - Case B: Verify pack should be installed
```

File: Tests/scripts/collect_tests_and_content_packs.py (hunk 1 above)
Comment (abaumgarten) [reply Tests/scripts/collect_tests_and_content_packs.py:1034]: Done

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 4]
```diff
@@ -1030,6 +1029,21 @@ def filter_tests(tests: set, id_set: json) -> set:
...
+        - Content pack is not in skipped packs
+        - Content pack is certified
```
[the remark this answers is not in the document]
Comment (abaumgarten) [reply Tests/scripts/collect_tests_and_content_packs.py:1036]: Done

File: Tests/scripts/collect_tests_and_content_packs.py [hunk 5]
```diff
@@ -1030,6 +1029,21 @@ def filter_tests(tests: set, id_set: json) -> set:
...
+        - Content pack is not deprecated
+    Args:
+        packs_to_install (set): Set of installed packs collected so far.
+    Returns:
+        (set): Set of packs without ignored, skipped and deprecated-packs.
+    """
+
+    return {pack for pack in packs_to_install if should_install_content_pack(pack)}
```
[the remark this answers is not in the document]
Comment (abaumgarten) [reply Tests/scripts/collect_tests_and_content_packs.py:1044]: Done

File: Tests/scripts/utils/content_packs_util.py [hunk 6]
```diff
@@ -75,3 +76,27 @@ def should_test_content_pack(pack_name: str) -> Tuple[bool, str]:
     if is_pack_deprecated(pack_path):
         return False, 'Pack is Deprecated'
     return True, ''
+
+
+def should_install_content_pack(pack_name: str) -> Tuple[bool, str]:
+    """Checks if content pack should be installed:
+        - Content pack is not in skipped packs
+        - Content pack is certified
```
[the remark this answers is not in the document]
Comment (abaumgarten) [reply Tests/scripts/utils/content_packs_util.py:84]: Done

File: Tests/scripts/utils/tests/content_packs_util_test.py (hunk 2 above)
Comment (abaumgarten) [reply Tests/scripts/utils/tests/content_packs_util_test.py:131]: Done

File: Tests/scripts/utils/tests/content_packs_util_test.py (hunk 3 above)
Comment (abaumgarten) [reply Tests/scripts/utils/tests/content_packs_util_test.py:135]: Done

**Comment by ShahafBenYakir** [comment]
Can you also deprecate a pack to show that it is till ignore those?

**Comment by abaumgarten** [comment]
> Can you also deprecate a pack to show that it is till ignore those?

Here you can see that `Alexa` was not included as it was changed to be a deprecated pack.
https://app.circleci.com/pipelines/github/demisto/content/66545/workflows/8ffbf00e-8660-4c4b-8e6a-83b751644d7d/jobs/274714

**Review by ShahafBenYakir — COMMENTED** [review]
I am not sure I understand the changes you did in `test_collect_tests_and_content_packs`. Each of the tested packs (FakePack,GreatPack) seems like wasn't collected, isn't this a bug?

**Comment by abaumgarten** [comment]
> I am not sure I understand the changes you did in `test_collect_tests_and_content_packs`. Each of the tested packs (FakePack,GreatPack) seems like wasn't collected, isn't this a bug?

Those two should not be collected as installed packs because they both recognized as deprecated packs (which we remove from installed packs).

File: Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py [hunk 7]
```diff
@@ -216,13 +217,22 @@ class TestChangedPlaybook:
...
         mocker.patch.object(Tests.scripts.collect_tests_and_content_packs, 'should_test_content_pack',
                             return_value=(True, ''))
         filterd_tests, content_packs = get_mock_test_list(git_diff_ret=self.GIT_DIFF_RET)
 
         assert filterd_tests == {self.TEST_ID}
-        assert content_packs == {"Base", "DeveloperTools", "CommonPlaybooks", "FakePack"}
+        assert "FakePack" not in content_packs
+        assert content_packs == {"Base", "DeveloperTools", "CommonPlaybooks"}
 
+        # pack FakePack is not deprecated and should be collected as an installed pack
```
Comment (ShahafBenYakir) [inline Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py:229]: Let's just seperate the two scenarios into two functions with clear namings

File: Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py [hunk 8]
```diff
@@ -400,13 +410,23 @@ class TestChangedIntegrationAndPlaybook:
...
         mocker.patch.object(Tests.scripts.collect_tests_and_content_packs, 'should_test_content_pack',
                             return_value=(True, ''))
         filterd_tests, content_packs = get_mock_test_list(git_diff_ret=self.GIT_DIFF_RET)
 
         assert filterd_tests == set(self.TEST_ID.split('\n'))
-        assert content_packs == {"Base", "DeveloperTools", 'CommonPlaybooks', 'PagerDuty', 'FakePack'}
+        assert "FakePack" not in content_packs
+        assert content_packs == {"Base", "DeveloperTools", 'CommonPlaybooks', 'PagerDuty'}
 
+        # Pack FakePack is not deprecated and should be in collected as an installed pack
```
Comment (ShahafBenYakir) [inline Tests/scripts/infrastructure_tests/test_collect_tests_and_content_packs.py:422]: same here

File: Tests/scripts/utils/content_packs_util.py [hunk 9]
```diff
@@ -75,3 +76,26 @@ def should_test_content_pack(pack_name: str) -> Tuple[bool, str]:
...
+        - Content pack is not in skipped packs
+        - Content pack is not deprecated
+
+    Args:
+        pack_name (str): The pack name to check if it should be tested
+
+    Returns:
+        bool: True if should be installed, False otherwise
+    """
+    if not pack_name:
```
[the remark this answers is not in the document]
Comment (abaumgarten) [reply Tests/scripts/utils/content_packs_util.py:92]: I kept the same logic as in the function `should_test_content_pack`,  I assumed it should be the same

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**